commit 899206bf8cd8bd5f42ce5ab40db71cf67414fb50
Author: Richard Imaoka <richard.e.imaoka@gmail.com>
Date:   Sun May 8 09:53:23 2022 +0900

    Add ApolloClient to App.tsx

diff --git a/client/src/components/App.tsx b/client/src/components/App.tsx
index 291b82d..7fb54b0 100644
--- a/client/src/components/App.tsx
+++ b/client/src/components/App.tsx
@@ -1,10 +1,16 @@
+import { ApolloProvider, ApolloClient, InMemoryCache } from "@apollo/client";
 import { HeaderContainer } from "./header/HeaderContainer";
 
+const client = new ApolloClient({
+  uri: "http://localhost:4000",
+  cache: new InMemoryCache(),
+});
+
 function App() {
   return (
-    <>
+    <ApolloProvider client={client}>
       <HeaderContainer />
-    </>
+    </ApolloProvider>
   );
 }
 
