commit d389589c248c2d3d1ea06c22778a56978ce88eb3
Author: Richard Imaoka <richard.e.imaoka@gmail.com>
Date:   Sun May 8 09:51:21 2022 +0900

    simplify App.tsx

diff --git a/client/src/App.tsx b/client/src/App.tsx
index 9cc63d9..db00f63 100644
--- a/client/src/App.tsx
+++ b/client/src/App.tsx
@@ -1,26 +1,7 @@
 import React from "react";
-import logo from "./logo.svg";
-import "./App.css";
 
 function App() {
-  return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.tsx</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
-  );
+  return <div>empty react application</div>;
 }
 
 export default App;
