commit fc588739ad5537fe27d2e76d6b128d04cc553a07
Author: Richard Imaoka <richard.e.imaoka@gmail.com>
Date:   Sun May 8 10:41:40 2022 +0900

    update DivisionCard.tsx to take generated DivisionCardFragment

diff --git a/client/src/components/company/DivisionCard.tsx b/client/src/components/company/DivisionCard.tsx
index 0ccd864..05690af 100644
--- a/client/src/components/company/DivisionCard.tsx
+++ b/client/src/components/company/DivisionCard.tsx
@@ -1,7 +1,12 @@
 import { gql } from "@apollo/client";
+import { DivisionCardFragment } from "../../generated/graphql";
 
-export const DivisionCard = (): JSX.Element => {
-  return <div></div>;
+export interface DivisionCardProps {
+  fragment: DivisionCardFragment;
+}
+
+export const DivisionCard = ({ fragment }: DivisionCardProps): JSX.Element => {
+  return <div>{fragment.divisionName}</div>;
 };
 
 DivisionCard.fragment = gql`
